<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Pong</class>
 <widget class="QMainWindow" name="Pong">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>450</width>
    <height>650</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>450</width>
    <height>650</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>450</width>
    <height>650</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>P2Pong</string>
  </property>
  <property name="styleSheet">
   <string notr="true">background-color: rgb(84, 85, 86);</string>
  </property>
  <widget class="QWidget" name="wCenterizer">
   <layout class="QVBoxLayout" name="verticalLayout">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QFrame" name="frame">
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Plain</enum>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="DraggableWidget" name="wTopBar" native="true">
         <layout class="QHBoxLayout" name="topbar">
          <property name="spacing">
           <number>6</number>
          </property>
          <property name="leftMargin">
           <number>3</number>
          </property>
          <property name="topMargin">
           <number>3</number>
          </property>
          <property name="rightMargin">
           <number>3</number>
          </property>
          <property name="bottomMargin">
           <number>3</number>
          </property>
          <item>
           <widget class="QPushButton" name="btnOptions">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="styleSheet">
             <string notr="true">#btnOptions {
	font: 9pt &quot;Segoe Script&quot;;
	background-color: rgb(148, 148, 148);
	padding: 4px;
	border-radius: 10px;
}
#btnOptions:pressed {
	background-color: rgb(120, 120, 120);
}</string>
            </property>
            <property name="text">
             <string>Options</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="lUsername">
            <property name="layoutDirection">
             <enum>Qt::LeftToRight</enum>
            </property>
            <property name="styleSheet">
             <string notr="true">font: 9pt &quot;Segoe Script&quot;;</string>
            </property>
            <property name="text">
             <string>username: &lt;null&gt;</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
            <property name="margin">
             <number>5</number>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QStackedWidget" name="swWindowsStack">
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="currentIndex">
          <number>0</number>
         </property>
         <widget class="QWidget" name="pwMainMenu">
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>16777215</height>
           </size>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <property name="spacing">
            <number>15</number>
           </property>
           <property name="sizeConstraint">
            <enum>QLayout::SetDefaultConstraint</enum>
           </property>
           <item>
            <widget class="QLabel" name="lTitle">
             <property name="enabled">
              <bool>true</bool>
             </property>
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="sizeIncrement">
              <size>
               <width>0</width>
               <height>0</height>
              </size>
             </property>
             <property name="styleSheet">
              <string notr="true">font: 700 72pt &quot;Segoe Script&quot;;</string>
             </property>
             <property name="text">
              <string>P2Pong</string>
             </property>
             <property name="scaledContents">
              <bool>false</bool>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
             <property name="textInteractionFlags">
              <set>Qt::NoTextInteraction</set>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btnGenCode">
             <property name="sizePolicy">
              <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="maximumSize">
              <size>
               <width>414</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="cursor">
              <cursorShape>PointingHandCursor</cursorShape>
             </property>
             <property name="styleSheet">
              <string notr="true">#btnGenCode {
	font: 24pt &quot;Segoe Script&quot;;
	background-color: rgb(148, 148, 148);
	padding: 10px;
	border-radius: 10px;
}
#btnGenCode:pressed {
	background-color: rgb(120, 120, 120);
}</string>
             </property>
             <property name="text">
              <string>Generate Code</string>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="layCode">
             <property name="sizeConstraint">
              <enum>QLayout::SetMaximumSize</enum>
             </property>
             <item>
              <widget class="QLabel" name="lCode">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="sizeIncrement">
                <size>
                 <width>0</width>
                 <height>0</height>
                </size>
               </property>
               <property name="contextMenuPolicy">
                <enum>Qt::NoContextMenu</enum>
               </property>
               <property name="styleSheet">
                <string notr="true">font: 36pt &quot;Segoe Script&quot;;</string>
               </property>
               <property name="text">
                <string/>
               </property>
               <property name="alignment">
                <set>Qt::AlignCenter</set>
               </property>
               <property name="textInteractionFlags">
                <set>Qt::LinksAccessibleByMouse|Qt::TextSelectableByMouse</set>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btnCopyCode">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="minimumSize">
                <size>
                 <width>100</width>
                 <height>100</height>
                </size>
               </property>
               <property name="sizeIncrement">
                <size>
                 <width>0</width>
                 <height>0</height>
                </size>
               </property>
               <property name="cursor">
                <cursorShape>PointingHandCursor</cursorShape>
               </property>
               <property name="toolTip">
                <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Copy Code&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
               </property>
               <property name="autoFillBackground">
                <bool>false</bool>
               </property>
               <property name="styleSheet">
                <string notr="true">#btnCopyCode {
	image: url(:/copy-96.png);
	border: none;
	background-repeat: none;
}
#btnCopyCode:pressed {
	image: url(:/copy-clicked-96.png);
}</string>
               </property>
               <property name="text">
                <string/>
               </property>
               <property name="icon">
                <iconset>
                 <normaloff>copy-96.png</normaloff>copy-96.png</iconset>
               </property>
               <property name="iconSize">
                <size>
                 <width>96</width>
                 <height>96</height>
                </size>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="layInput">
             <property name="sizeConstraint">
              <enum>QLayout::SetDefaultConstraint</enum>
             </property>
             <item>
              <spacer name="horizontalSpacer">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item>
              <widget class="AutoClipLineEdit" name="lineEdit">
               <property name="minimumSize">
                <size>
                 <width>210</width>
                 <height>0</height>
                </size>
               </property>
               <property name="maximumSize">
                <size>
                 <width>210</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="styleSheet">
                <string notr="true">font: 24pt &quot;Segoe Script&quot;;
border: 4px solid;
border-radius: 10px;
padding: 5px;
border-color: rgb(120, 120, 120);</string>
               </property>
               <property name="maxLength">
                <number>6</number>
               </property>
               <property name="frame">
                <bool>false</bool>
               </property>
               <property name="alignment">
                <set>Qt::AlignCenter</set>
               </property>
               <property name="placeholderText">
                <string>Code here</string>
               </property>
               <property name="clearButtonEnabled">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btnConnect">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="sizeIncrement">
                <size>
                 <width>0</width>
                 <height>0</height>
                </size>
               </property>
               <property name="cursor">
                <cursorShape>PointingHandCursor</cursorShape>
               </property>
               <property name="mouseTracking">
                <bool>false</bool>
               </property>
               <property name="styleSheet">
                <string notr="true">#btnConnect {
	font: 24pt &quot;Segoe Script&quot;;
	background-color: rgb(148, 148, 148);
	padding: 10px;
	border-radius: 10px;
}
#btnConnect:pressed {
	background-color: rgb(120, 120, 120);
}</string>
               </property>
               <property name="text">
                <string>Connect</string>
               </property>
               <property name="iconSize">
                <size>
                 <width>16</width>
                 <height>16</height>
                </size>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer_2">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="pwPongWindow">
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <widget class="PongView" name="graphicsView"/>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>DraggableWidget</class>
   <extends>QWidget</extends>
   <header>draggablewidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>AutoClipLineEdit</class>
   <extends>QLineEdit</extends>
   <header>autocliplineedit.h</header>
  </customwidget>
  <customwidget>
   <class>PongView</class>
   <extends>QGraphicsView</extends>
   <header>pongview.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>btnOptions</tabstop>
  <tabstop>graphicsView</tabstop>
  <tabstop>lineEdit</tabstop>
  <tabstop>btnConnect</tabstop>
  <tabstop>btnGenCode</tabstop>
  <tabstop>btnCopyCode</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>lineEdit</sender>
   <signal>returnPressed()</signal>
   <receiver>btnConnect</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>189</x>
     <y>551</y>
    </hint>
    <hint type="destinationlabel">
     <x>325</x>
     <y>560</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
